ruby/command-t/lib/command-t/vim.rb

```diff
@@ -81,12 +81,12 @@ def wildignore_to_regexp(str)
           elsif pattern.match(%r{\A\*\.([^*]+)\z})
             # *.something (match file with extension at any level)
             '\.' + Regexp.escape($~[1]) + '\z'
+          elsif pattern.match(%r{\A\*/([^/]+)/\*\z})
+            # */something/* (match directories at any level)
+            '(\A|/)' + Regexp.escape($~[1]) + '/.*(/|\z)'
```

Comment:
I have a feeling we can combine the `*/foo` and `*/foo/*` patterns into one, but I'll leave that to a future effort.